metaflow/plugins/datatools/s3/s3.py

```diff
@@ -600,7 +600,7 @@ def _url(self, key_value):
         # returned are Unicode.
         key = getattr(key_value, "key", key_value)
         if self._s3root is None:
-            parsed = urlparse(to_unicode(key))
+            parsed = urlparse(to_unicode(key), allow_fragments=False)
```

Comment:
can you add a comment here calling out why `allow_fragments` needs to be set to False for posterity